src/coreclr/jit/fgbasic.cpp

```diff
@@ -3252,7 +3252,13 @@ void Compiler::fgFindBasicBlocks()
         // and clear the rarely run flag
         hndBegBB->makeBlockHot();
 #else
-        hndBegBB->bbSetRunRarely();   // handler entry points are rarely executed
+        // Handler entry points are rarely executed,
+        // but don't assume finally clause is cold
+        // (we want to enable fgCloneFinally() optimization).
+        if (hndBegBB->bbJumpKind != BBJ_EHFINALLYRET)
+        {
+            hndBegBB->bbSetRunRarely();
```

Comment:
Why do we make any sort of assumption here? 

Seems like perhaps we should just leave `hndBegBB` alone in this case.

Reply:
As far as I can tell, this assumption may be useful for selecting a splitting point in `fgDetermineFirstColdBlock`, but this hasn't been observed since `HANDLER_ENTRY_MUST_BE_IN_HOT_SECTION` has been forcing handler entries to be hot. Without this assumption, I don't think a handler entry will ever be selected as a candidate for `fgFirstColdBlock` [here](https://github.com/dotnet/runtime/blob/1c8ea7081914543f166c507f778267f8eb3c4d50/src/coreclr/jit/flowgraph.cpp#L3462).

Reply:
> Seems like perhaps we should just leave `hndBegBB` alone in this case.

@AndyAyersMS Do you think that here, and for the `filtBB->bbSetRunRarely()` case below, if we're not forcing the weight, we should just leave it alone and let natural weights be applied? In particular, we don't want to overwrite profile collected weights, which this is doing. It seems like the right thing to do would be to add logic to `fgComputeMissingBlockWeights()`, which computes weights for blocks that don't have any profile weights, so for every handler we call `bbSetRunRarely()` there.

Reply:
Something like that.

In the absence of PGO, I think we can just decide catch / filter / fault funclets are rare with impunity (since they are only reached if there is an exception), but finallys should have a weight similar to the callfinallys that can invoke them.